backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py

```diff
@@ -186,6 +186,78 @@ def delete_indices(self) -> None:
                 # Ignore error when no matching alias exists
                 pass
 
+    def delete_indice(self, content: Content) -> None:
+        """
+        Delete a content from elastic_search engine
+        """
+        content_in_context = ContentInContext(content, config=self._config, dbsession=self._session)
+        logger.info(self, "Delete content {} index".format(content_in_context.content_id))
+        author = self._create_digest_user_from_user(content_in_context.author)
+        last_modifier = self._create_digest_user_from_user(content_in_context.last_modifier)
+        workspace = DigestWorkspace(
+            workspace_id=content_in_context.workspace.workspace_id,
+            label=content_in_context.workspace.label,
+        )
+        path = [
+            DigestContent(
+                content_id=component.content_id,
+                label=component.label,
+                slug=component.slug,
+                content_type=component.content_type,
+            )
+            for component in content_in_context.content_path
+        ]
+        indexed_subcontent = self._index_subcontents(content_in_context)
+        comments = indexed_subcontent["comments"]
+        todos = indexed_subcontent["todos"]
```

Comment:
Should use `indexed_subcontent.get(...)`. This will return a default value instead of raising a `KeyError` exception (and you can provide a default value too). Unless raising an exception is desired behavior.

Reply:
indexed_subcontent is created specifically to get the comments and todos, the value aren't supposed to have wrong value, raising an error in that case seems good

Reply:
ok for me